ethexe/observer/src/lib.rs

```diff
@@ -147,22 +181,37 @@ impl FusedStream for ObserverService {
 }
 
 impl ObserverService {
-    pub async fn new(config: &EthereumConfig) -> Result<Self> {
-        let blobs = Arc::new(
-            ConsensusLayerBlobReader::new(&config.rpc, &config.beacon_rpc, config.block_time)
-                .await
-                .context("failed to create blob reader")?,
-        );
-
-        Self::new_with_blobs(config, blobs).await
-    }
+    pub async fn new<DB>(
+        eth_cfg: &EthereumConfig,
+        max_sync_depth: u32,
+        db: &DB,
+        blobs_reader: Option<Arc<dyn BlobReader>>,
```

Comment:
such options always look strange, especially if in most of places we use Some. I'd keep it splitted into two fns

Reply:
Yes, but I think we should create BlobReader outside always. And provide it to observer after.

Reply:
Why? There're some default configurations we support (as ConsensusBlobReader)

Reply:
1) You can see in the code - each time we creates observer we provides blob reader argument 
2) Maybe in future blob reader can support not only receive blobs using RPC requests, but also request blobs from vara or from p2p.

Reply:
ok, lets always pass it

Reply:
Ok, todo is added and hope @ecol-master will do it